### ISSUE TITLE
ADOLoadCode("postgres") fails.

### ISSUE THREAD

**Issue by WiXSL** [ADOdb/ADOdb#135]
ADODB V5.19
File: adodb/adodb.inc.php

When ADODB calls ADOLoadCode() with "postgres" as argument, $class variable is set to "postgres8" ("$class = $db = 'postgres8';") which generate the evaluation of the existence of the class "ADODB_postgres8" that doesn't exist! ("if (class_exists("ADODB_" . $class)) return $class;")
and getting an error (ADOConnection::outp("Syntax error in file: $file");) since file "adodb-postgres8.inc.php" is empty.

	function ADOLoadCode($dbType)
	{
	global $ADODB_LASTDB;

		if (!$dbType) return false;
		$db = strtolower($dbType);
		switch ($db) {
			case 'ado':
				if (PHP_VERSION >= 5) $db = 'ado5';
				$class = 'ado';
				break;

			case 'ifx':
			case 'maxsql':
				$class = $db = 'mysqlt';
				break;

			case 'pgsql':
			case 'postgres':

            /* BUG: I change the following line to: $class = $db = 'postgres7'; */

				$class = $db = 'postgres8';
				break;

			default:
				$class = $db; break;
		}

		$file = ADODB_DIR."/drivers/adodb-".$db.".inc.php";
		@include_once($file);
		$ADODB_LASTDB = $class;
		if (class_exists("ADODB_" . $class)) return $class;

		//ADOConnection::outp(adodb_pr(get_declared_classes(),true));
		if (!file_exists($file)) ADOConnection::outp("Missing file: $file");
		else ADOConnection::outp("Syntax error in file: $file");
		return false;
	}

**Comment by mnewnham** [comment]
Could you give a little more context please, as class ADODB_postgres8 oviously does exist (drivers/adodb-postgres8.inc.php)

**Comment by mnewnham** [comment]
To expand a little on my comment above. There is no postgres driver as such. Your connection statement should reference 'postgres64,  'postgres7', 'postgres8' or 'postgres9' depending on the version of the database

**Comment by dregad** [comment]
> Your connection statement should reference 'postgres64, 'postgres7', 'postgres8' or 'postgres9' depending on the version of the database

Actually, this is not 100% accurate - `ADONewConnection()` calls `ADOLoadCode()` which (currently) sets the driver name to *postgres8* when given *postgres* or *pgsql*, allowing to use the "default" driver instead of a specific one.

**Comment by mnewnham** [comment]
So actually my 'expansion' was more incorrect than my original comment. @WiXSL Could you clarify, what you mean by 'empty', as the drivers/adodb-postgres8.inc.php should simply load the postgres7 driver. @dregad , I believe we should flag driver-specific workarounds  in the core programs for removal in the next major release

**Comment by dregad** [comment]
> flag driver-specific workarounds in the core programs for removal in the next major release

agreed.

**Comment by WiXSL** [comment]
Ok,
We have always use 'postgres' as a parameter to access postgres 7,8,9 databases with our projects using adodb V5.11 witch the "drivers/adodb-postgres8.inc.php" driver had the following content:

<?php
/*
 V5.11 5 May 2010   (c) 2000-2010 John Lim (jlim#natsoft.com). All rights reserved.
  Released under both BSD license and Lesser GPL library license. 
  Whenever there is any discrepancy between the two licenses, 
  the BSD license will take precedence.
  Set tabs to 4.
  
  NOTE: The "postgres8" driver is remapped to "postgres7". 
*/
?>

Recently we upgraded to V5.19 and apparently that file didn't get correctly upgraded for some reason (team miss-communication).
That was causing the error.

Really Sorry for the confusion.

At least i can see that help for considering flag driver-specific workarounds idea.
